tools/kwiver_tool_runner.cxx

```diff
@@ -22,8 +21,93 @@
 #include <memory>
 #include <utility>
 
-using applet_factory = kwiver::vital::implementation_factory_by_name< kwiver::tools::kwiver_applet >;
 using applet_context_t = std::shared_ptr< kwiver::tools::applet_context >;
+namespace kv = kwiver::vital;
+
+class command_node;
+using command_node_sptr = std::shared_ptr<command_node>;
+bool compare_node( command_node_sptr const& n1, command_node_sptr const& n2 );
+
+// -------------------------------------------------------------------
+class command_node
+{
+public:
+  command_node(std::string const& name)
+    : m_name( name ),
+      m_is_command( false )
+  { }
+
+  virtual ~command_node() = default;
+
+  void
+  add_factory( kv::plugin_factory_handle_t fact )
+  {
+    m_factory = fact;
+    m_is_command = true;
+  }
+
+  bool
+  is_command() const { return m_is_command; }
+
+  command_node_sptr
+  find( std::string const& name )
+  {
+    for ( auto& n : m_nodes )
+    {
+      if ( name == n->m_name )
+      {
+        return n;
+      }
+    } // end for
+    return nullptr;
+  }
+
+  void
+  add( command_node_sptr n )
+  {
+    m_nodes.push_back(n);
+
+    // sort vector
```

Comment:
this comment is unnecessary as is, but you could explain why this needs to be sorted.  Alternatively you could just use a data structure like `std::set` that keeps them sorted.

Reply:
Done